### PR TITLE
baby steps on patching inf/nan behavior & aten::amin support in nvfuser

### LINKED ISSUE #75622: NVFuser incorrectly computes max for extremal values
```
In [2]: import torch

In [3]: def fn(x):
   ...:     x=x.amax(dim=0)*.1
   ...:     return x
   ...: 
   ...: a=torch.tensor([-float('inf'), -float('inf'), -float('inf'), -float('inf')], device="cuda")
   ...: scripted = torch.jit.script(fn)
   ...: fn(a)
   ...: with torch.jit.fuser("fuser2"):
   ...:     for _ in range(10):
   ...:         out = scripted(a)
   ...: print(out)
tensor(-3.4028e+37, device='cuda:0')
```

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes #75622 

1. Instead of getting max/min_value for reduction init value, we go with (-)infinity instead so we can properly preserve inf inputs;
2. Adding inf/(-)inf/nan for float value.
3. Adding aten::amin in nvfuser (@kevinstephano @rdspring1 for review)

### CODE REVIEW

**Comment by jjsjann123** [comment]
cc'ing @kevinstephano

File: test/test_jit_cuda_fuser.py [hunk 1]
```diff
@@ -4393,6 +4393,27 @@ def f(x):
         self.assertEqual(prof.events().table().find("fallback"), -1)
         torch._C._jit_set_nvfuser_guard_mode(old_guard)
 
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    @unittest.skipIf(is_pre_volta(), "reduction not supported in pre volta device")
+    def test_inf_quick_patch(self):
+        x = torch.tensor([-float('inf'), -float('inf'), 4.0], device="cuda")
```
Comment (ngimel) [inline test/test_jit_cuda_fuser.py:4401]: This test is not enough to catch previously problematic values - you need input full of -inf (nothing else there) to catch that FLT_MIN is not the correct initializer, similarly for amin. You need yet another input for nan propagation (unless you have those tests in other places, in which case it's probably better to unify)

**Review by rdspring1 — APPROVED** [review]
LGTM

**Review by kevinstephano — APPROVED** [review]
LGTM

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**